admin-dev/themes/new-theme/js/pages/discount/form/specific-products.ts

```diff
@@ -0,0 +1,48 @@
+/**
+ * Copyright since 2007 PrestaShop SA and Contributors
+ * PrestaShop is an International Registered Trademark & Property of PrestaShop SA
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.md.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://devdocs.prestashop.com/ for more information.
+ *
+ * @author    PrestaShop SA and Contributors <contact@prestashop.com>
+ * @copyright Since 2007 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+*/
+
+import {EventEmitter} from 'events';
+import EntitySearchInput from '@components/entity-search-input';
+import DiscountMap from '@pages/discount/discount-map';
+
+const {$} = window;
+export default class SpecificProducts {
```

Comment:
We'll have to discuss how we handle this JS code, I'm afraid one class per form element will end up with too many split code that doesn't help with code readability It's totally ok for this PR though, and can be discussed later